### PR TITLE
Deprecate setting verbose for individual commands

### PR DESCRIPTION
**Pull request by omus**
Instead move towards having verbose set on a connection basis.

### CODE REVIEW

**Comment by omus** [comment]
I'll have one more commit which will remove the use of the deprecations in the tests

File: src/conn_context.jl [hunk 1]
```diff
@@ -110,16 +114,24 @@ Download a file with a persistent connection. Returns a `Response`.
...
     save_path::AbstractString="";
     mode::FTP_MODE=binary_mode,
-    verbose::Union{Bool,IOStream}=false,
+    verbose=nothing,
 )
+    if verbose !== nothing
+        Base.depwarn(
+            string(
+                "The `verbose` keyword is deprecated and now needs to be supplied during ",
+                "the initialization of `FTP`.",
```
Comment (omus) [inline src/conn_context.jl]: Oh shoot, I need to update this...

File: src/FTPObject.jl [hunk 2]
```diff
@@ -136,8 +135,18 @@ function download(
...
 )
+    if verbose !== nothing
+        Base.depwarn(
+            string(
+                "The `verbose` keyword is deprecated and now needs to be supplied during ",
+                "the initialization of `FTP`.",
+            ),
+            :download,
+        )
+    end
```
Comment (ararslan) [inline src/FTPObject.jl]: You're writing this a lot, so I'd refactor it out into a function:
```julia
function _check_verbose(f, value)
    if verbose !== nothing
        Base.depwarn("The `verbose` keyword is deprecated and now needs to be supplied " *
                     "during the initialization of `FTP`.", nameof(f))
    end
end
```
Then in each of these functions, simply write e.g. `_check_verbose(download, verbose)`.

Comment (omus) [reply src/FTPObject.jl]: You're right

**Review by ararslan — APPROVED** [review]

**Pull request merged**